File: video-processor/modules/exceptions/TokenNotFoundException.js

const {StatusCodes} = require("http-status-codes");
const {AutomaticallyHandledException} = require("../global-route-exceptions-handler/exceptions/AutomaticallyHandledException");


class TokenNotFoundException extends AutomaticallyHandledException{
    constructor(tokenName) {
        super("Token not found", StatusCodes.NOT_FOUND, {
            token: tokenName
        });
        this.name = "TokenNotFoundException";
    }
}
module.exports.TokenNotFoundException = TokenNotFoundException;

File: export_collection_orchestrations/export_collections/exceptions/TokenNotFoundException.py
from http import HTTPStatus

from global_exception.exceptions.AutomaticallyHandledException import AutomaticallyHandledException


class TokenNotFoundException(AutomaticallyHandledException):
    def __init__(self, token):
        super().__init__(
            HTTPStatus.BAD_REQUEST,
            f"Token {token} is not found",
            {'token': token, }
        )
